lib/spack/spack/build_environment.py

```diff
@@ -577,8 +577,14 @@ def child_execution(child_connection, input_stream):
     parent_connection, child_connection = multiprocessing.Pipe()
     try:
         # Forward sys.stdin to be able to activate / deactivate
-        # verbosity pressing a key at run-time
-        input_stream = lang.duplicate_stream(sys.stdin)
+        # verbosity pressing a key at run-time.  When sys.stdin can't
+        # be duplicated (e.g. running under nohup, which results in an
+        # '[Errno 22] Invalid argument') then just use os.devnull
+        try:
+            input_stream = lang.duplicate_stream(sys.stdin)
+        except:
```

Comment:
Turns out that it's an OSError, not an IOError, but otherwise I think that this is what you're looking for.